### ISSUE TITLE
check how checkpoints are handled

### ISSUE THREAD

**Issue by alecristia** [LAAC-LSCP/pyannote-vtc-testing#1]
dixit : 
Yes, it's entirely possible that `SimpleSegmentationModel`, `PyanNet`, or the training framework you're using, such as PyTorch Lightning, already handle model checkpointing as part of their internal mechanisms or default configurations. Here's how these components might manage model saving:

### PyTorch Lightning's Automatic Checkpointing
PyTorch Lightning offers built-in support for automatic checkpointing through its `ModelCheckpoint` callback, which can be configured to save model checkpoints at specified intervals, upon improvements in a monitored metric (like validation loss), or at the end of training. If you've added a `ModelCheckpoint` callback to your `Trainer`, PyTorch Lightning takes care of saving the model based on the conditions you've specified.

### `SimpleSegmentationModel` and `PyanNet`
If `SimpleSegmentationModel` or `PyanNet` are classes from the `pyannote.audio` library or another source, they might come with pre-defined training and saving behaviors, especially if they are designed to integrate seamlessly with PyTorch Lightning or another training framework. This could include:
- Pre-defined callbacks for model checkpointing.
- Methods that automatically save the model's best version during training.

### Checking for Built-in Checkpointing
To determine whether `SimpleSegmentationModel`, `PyanNet`, or your training setup automatically handles model checkpointing, consider the following steps:
1. **Review the Documentation**: Check the official documentation for `pyannote.audio`, PyTorch Lightning, or the source of your model classes for information on default behaviors regarding model saving and checkpointing.

2. **Inspect the Class Definitions**: Look at the source code of `SimpleSegmentationModel` and `PyanNet` to see if they define any model saving logic or configure any `ModelCheckpoint` callbacks internally.

3. **Configuration Inspection**: If you're configuring a `Trainer` instance in PyTorch Lightning, review how you set up the `ModelCheckpoint` callback and other callbacks to ensure they're correctly configured to save your model as desired.

### Conclusion
It's quite plausible that model checkpointing is being handled automatically through the mechanisms provided by PyTorch Lightning or the specific model classes you're using. Ensuring you have correctly set up and configured these mechanisms will help you achieve the desired model saving behavior without needing to explicitly call a save method at the end of training.

If automatic checkpointing meets your needs, you might not need to add any additional model saving code. However, if you have specific requirements for model saving that aren't covered by the default behavior, you may still want to implement explicit saving logic as previously discussed.

Consider adding:
```
from pytorch_lightning.callbacks import ModelCheckpoint

model_checkpoint = ModelCheckpoint(
    monitor='val_loss',  # Specify what to monitor, e.g., validation loss
    dirpath='checkpoints/',  # Directory where checkpoints are saved
    filename='model-{epoch:02d}-{val_loss:.2f}',
    save_top_k=1,  # Save only the best model
    mode='min',  # 'min' for minimizing the monitored value, 'max' for maximizing
)

trainer_kwargs['callbacks'].append(model_checkpoint)

```
